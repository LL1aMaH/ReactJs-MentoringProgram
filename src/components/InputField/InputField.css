.warning {
  position: absolute;
  font-size: 0.875rem;
  line-height: 1rem;
  border-radius: 4px;
  color: var(--white-color);
  background-color: var(--error-main-color);
  padding: 0.25rem 0.375rem 0.25rem 0.5rem;
}
