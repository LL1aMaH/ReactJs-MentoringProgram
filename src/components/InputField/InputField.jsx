import React, { memo } from 'react';

import { Field } from 'formik';

import styles from '../../../styles/InputField.module.scss'

export const InputField = memo(function InputField(props) {
  const {
    className,
    label,
    labelClassName,
    inputClassName,
    disabled,
    type = text,
    error,
    ...restProps
  } = props;

  return (
    <div className={className}>
      {error ? <div className={styles.warning}>{error}</div> : null}
      <span className={labelClassName}>{label}</span>
      <Field
        {...restProps}
        type={type}
        className={inputClassName}
        disabled={disabled}
        autoComplete="off"
      />
    </div>
  );
});
