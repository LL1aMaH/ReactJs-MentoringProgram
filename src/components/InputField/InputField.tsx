import React, { memo } from 'react';

export type InputFieldProps = {
  placeholder?: string;
  label?: string;
  className?: string;
  inputClassName?: string;
  labelClassName?: string;
  disabled?: boolean;
};

export const InputField = memo(function InputField(props: InputFieldProps): JSX.Element {
  const { placeholder, className, label, labelClassName, inputClassName, disabled } = props;
  return (
    <div className={className}>
      <span className={labelClassName}>{label}</span>
      <input type="text" placeholder={placeholder} className={inputClassName} disabled={disabled} />
    </div>
  );
});
