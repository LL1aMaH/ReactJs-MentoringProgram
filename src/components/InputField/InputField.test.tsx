import React from 'react';
import { render } from '@testing-library/react';

import getFormDecorator from '../../../utils/decorators/withForm';

import { InputField } from './InputField';

describe('InputField', () => {
  const withForm = getFormDecorator({ initialValues: { test: 'test' } });
  it('renders correctly', () => {
    const getComponent = () => render(withForm(() => <InputField />));

    expect(getComponent().asFragment()).toMatchSnapshot();
  });

  it('renders correctly with error', () => {
    const getComponent = () => render(withForm(() => <InputField error="error" />));

    expect(getComponent().asFragment()).toMatchSnapshot();
  });
});
