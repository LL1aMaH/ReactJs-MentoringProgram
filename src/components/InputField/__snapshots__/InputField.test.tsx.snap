// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<DocumentFragment>
  <div>
    <span />
    <input
      autocomplete="off"
      type="text"
      value="[object Object]"
    />
  </div>
</DocumentFragment>
`;

exports[`InputField renders correctly with error 1`] = `
<DocumentFragment>
  <div>
    <div
      class="warning"
    >
      error
    </div>
    <span />
    <input
      autocomplete="off"
      type="text"
      value="[object Object]"
    />
  </div>
</DocumentFragment>
`;
